src/ext/java/org/opentripplanner/ext/siri/SiriFuzzyTripMatcher.java

```diff
@@ -383,4 +385,41 @@ public int getTripArrivalTime(FeedScopedId tripId) {
         }
         return -1;
     }
+
+    /**
+     * Returns a match of tripIds that match the provided values.
+     *
+     * @param internalPlanningCode
+     * @param serviceDate
+     * @param mode
+     * @param transportSubmode
+     * @return
+     */
+    public List<FeedScopedId> getTripIdForInternalPlanningCodeServiceDateAndMode(String internalPlanningCode, ServiceDate serviceDate, TransitMainMode mode, String transportSubmode) {
+
+        Set<Trip> cachedTripsBySiriId = getCachedTripsBySiriId(internalPlanningCode);
+
+        if (cachedTripsBySiriId.isEmpty()) {
+            cachedTripsBySiriId = getCachedTripsByVehicleRef(internalPlanningCode);
+        }
+
+        List<FeedScopedId> matches = new ArrayList<>();
+        for (Trip trip : cachedTripsBySiriId) {
+            final TripPattern tripPattern = routingService.getPatternForTrip().get(trip);
+            if (tripPattern.getMode().getMainMode().equals(mode) ||
+                tripPattern.getMode().getSubMode().equals(transportSubmode)) {
```

Comment:
No null check here on `tripPattern.getMode().getSubMode()`? (There is a null sjek on this on line 229)

Reply:
Null check added